### PR TITLE
Backport PR #14707 on branch v5.0.x (Fix numpy-dev deprecation in model Parameter)

### PR DESCRIPTION
**Pull request by larrybradley**
Manual backport of #14707 to v5.0.x

### CODE REVIEW

File: astropy/modeling/tests/test_parameters.py [hunk 1]
```diff
@@ -543,11 +542,6 @@ def test_shape(self):
         param = Parameter(name="test", default=np.array([1]))
         assert param.shape == (1,)
         # Reshape error
-        if not NUMPY_LT_1_25:
-            # error message changed in numpy 1.25
-            MESSAGE = r"cannot reshape array of size 1"
-        else:
-            MESSAGE = r"Cannot assign this shape to a scalar quantity"
```
Comment (pllim) [inline astropy/modeling/tests/test_parameters.py:550]: I think we need to keep this `MESSAGE` or it is going to be undefined.

I am also going to close/reopen to make basebranch check happy.

**Comment by pllim** [comment]
p.s. I see 91 failures in v5.0.x branch now using numpy dev... maybe we're in a losing battle... but maybe this PR will fix those... 🤞 

https://github.com/astropy/astropy/actions/runs/4835228862/jobs/8617340816

File: astropy/modeling/tests/test_parameters.py (hunk 1 above)
Comment (pllim) [reply astropy/modeling/tests/test_parameters.py:550]: I pushed a commit to fix this. FYI.

**Comment by pllim** [comment]
`91 failed` 😱

**Comment by larrybradley** [comment]
Most of the failures come from the modeling `projections` module (which we already knew about).  The other few appear to be from the numpy sin/cos precision change.

**Comment by pllim** [comment]
I just went ahead and pin maxversion of numpy on v5.0.x

**Pull request closed without merging**